pkg/webhook/validator.go

```diff
@@ -26,9 +26,11 @@ import (
 	"github.com/google/go-containerregistry/pkg/authn/k8schain"
 	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sigstore/cosign/pkg/cosign"
 	"github.com/sigstore/cosign/pkg/oci"
 	ociremote "github.com/sigstore/cosign/pkg/oci/remote"
 	"github.com/sigstore/cosign/pkg/policy"
+	csigs "github.com/sigstore/cosign/pkg/signature"
```

Comment:
We've been trying (with the sigstore refactoring work that @imjasonh and others have been doing) to reduce dependencies to cosign and where ever possible to use sigstore/sigstore, so I'm not sure we want to add this one here. I can also see us maybe having additional error types that we define so maybe we could just have our Errors that we return just like you did in sigstore/cosign and use them here instead.

Reply:
I get that, but here it is a little chicken/egg.  I only added the error stuff to cosign because that's where the logic still lives (and that should be trivial to move to `sigstore/sigstore` when the signing/verification moves).

I'd have to look at what it would take to move the extension stuff over, but it'd be great to have a library in `sigstore/sigstore` that all of `cosign`, `fulcio`, and `policy-controller` can lean on 🤞

Reply:
Ack, that makes sense so that when the signing moves the errors would move as well. Hindsight, maybe the error stuff should've been put into sigstore/sigstore to begin with?

Reply:
Possibly, but I opted for the faster/lazier version of treating it as part of the verification chunk that'll move later 😏